sdlib/d/gc/tcache.d

```diff
@@ -54,6 +64,77 @@ public:
 		pd.arena.free(emap, pd, ptr);
 	}
 
+	/**
+	 * Appendables API.
+	 */
+
+	// Get the capacity of the array slice denoted by slice and length.
+	// It includes the space occupied by the slice itself.
+	size_t getArrayCapacity(void* sliceAddr, size_t sliceLen) {
```

Comment:
The GC has no notion of array, only allocation, and this is better this way, IMO. You have allocation, and you have the ability to extend some allocation to fit more in them. The rest doesn't have to be handled by the GC, but can be handled by the rest of the runtime (and is largely out of scope here).